src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/AppealSubmitFeePreparer.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.hmcts.reform.iacasepaymentsapi.domain.handlers.presubmit;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.iacasepaymentsapi.domain.entities.AsylumCaseDefinition.APPEAL_FEE_DESC;
+import static uk.gov.hmcts.reform.iacasepaymentsapi.domain.entities.AsylumCaseDefinition.APPEAL_TYPE;
+import static uk.gov.hmcts.reform.iacasepaymentsapi.domain.entities.AsylumCaseDefinition.FEE_AMOUNT;
+import static uk.gov.hmcts.reform.iacasepaymentsapi.domain.entities.AsylumCaseDefinition.ORAL_FEE_AMOUNT_FOR_DISPLAY;
+
+import java.util.Collections;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacasepaymentsapi.domain.entities.AppealType;
+import uk.gov.hmcts.reform.iacasepaymentsapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacasepaymentsapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacasepaymentsapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacasepaymentsapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacasepaymentsapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacasepaymentsapi.domain.entities.fee.Fee;
+import uk.gov.hmcts.reform.iacasepaymentsapi.domain.entities.fee.FeeType;
+import uk.gov.hmcts.reform.iacasepaymentsapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacasepaymentsapi.domain.service.FeeService;
+
+@Component
+public class AppealSubmitFeePreparer implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private final FeeService feeService;
+
+    private Fee fee;
+
+    public AppealSubmitFeePreparer(
+        FeeService feeService
+    ) {
+        this.feeService = feeService;
+    }
+
+    @Override
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_START
+               && callback.getEvent() == Event.SUBMIT_APPEAL;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final AsylumCase asylumCase =
+            callback
+                .getCaseDetails()
+                .getCaseData();
+
+        AppealType appealType = asylumCase.read(APPEAL_TYPE, AppealType.class)
+            .orElseThrow(() -> new IllegalStateException("AppealType is not present"));
+
+        if (appealType.equals(AppealType.EA)
+            || appealType.equals(AppealType.HU)
+            || appealType.equals(AppealType.PA)) {
+            if (!isFeeExists(FeeType.ORAL_FEE)) {
```

Comment:
instead of using field from object you can use `Fee fee = feeService.getFee(feeType)` here
and then if it is not null